people-and-planet-ai/land-cover-classification/serving/main.py

```diff
@@ -69,147 +64,28 @@ def predict(lat: float, lon: float, year: int) -> flask.Response:
 
     # Optional HTTP request parameters.
     #   https://en.wikipedia.org/wiki/Query_string
-    patch_size = flask.request.args.get("patch-size", 256, type=int)
-    model_path = flask.request.args.get("model-path", DEFAULT_MODEL_PATH)
+    patch_size = flask.request.args.get("patch-size", 512, type=int)
+    model_path = flask.request.args.get("model-path")
 
-    # If anything fails, send a more descriptive error message instead of
-    # a generic "500: Internal Server Error" message.
     try:
-        ee_init()
-
-        # Fetch the input patch.
-        patch = get_patch(
-            image=sentinel2_image(f"{year}-01-01", f"{year}-12-31"),
-            lat=lat,
-            lon=lon,
-            bands=INPUT_BANDS,
-            patch_size=patch_size,
-            scale=10,
-        )
-
-        # Load the model and get the predictions. If the model is hosted
-        # somewhere else, this is where we would send a request.
-        model = tf.keras.models.load_model(model_path)
-
-        # Create an input dictionary with a batch containing a single patch.
-        inputs_batch = {name: np.stack([patch[name]]) for name in patch.dtype.names}
-
-        # Get the first (and only) element in the predictions batch.
-        probabilities = model.predict(inputs_batch)[0]
+        # Load the model or use the default model.
+        model = tf.keras.models.load_model(model_path) if model_path else MODEL
```

Comment:
A fix would be to have a registry of models (Dict[str, Model])and have the client pass a key instead of the filepath.
This would also speed up the request, since you preload the model and store it in the dict during initialization.

Reply:
Good point, I just wanted to give flexibility but you're right (and mostly me testing out new models and not wanting to re-deploy). I'll simply remove this option.